code/modules/projectiles/pins.dm

```diff
@@ -229,84 +229,93 @@
 		owned = TRUE
 		payment_amount = transaction_amount
-		gun_owners += user
+		gun_owners += user.get_bank_account()
 		to_chat(user, span_notice("You link the card to the firing pin."))
 
 /obj/item/firing_pin/paywall/pin_auth(mob/living/user)
 	if(!istype(user))//nice try commie
 		return FALSE
 	var/datum/bank_account/credit_card_details = user.get_bank_account()
-	if(user in gun_owners)
+	if(credit_card_details in gun_owners)
 		if(multi_payment && credit_card_details)
+			if(!gun.can_shoot())
+				return TRUE //So you don't get charged for attempting to fire an empty gun.
 			if(credit_card_details.adjust_money(-payment_amount, "Firing Pin: Gun Rent"))
 				if(pin_owner)
-					pin_owner.registered_account.adjust_money(payment_amount, "Firing Pin: Payout For Gun Rent")
+					pin_owner.adjust_money(payment_amount, "Firing Pin: Payout For Gun Rent")
 				return TRUE
 			to_chat(user, span_warning("ERROR: User balance insufficent for successful transaction!"))
 			return FALSE
 		return TRUE
-	if(credit_card_details && !active_prompt)
-		var/license_request = tgui_alert(user, "Do you wish to pay [payment_amount] credit[( payment_amount > 1 ) ? "s" : ""] for [( multi_payment ) ? "each shot of [gun.name]" : "usage license of [gun.name]"]?", "Weapon Purchase", list("Yes", "No"))
+	if(credit_card_details)
+		if(active_prompt)
+			return FALSE
 		active_prompt = TRUE
+		var/license_request = tgui_alert(user, "Do you wish to pay [payment_amount] credit[( payment_amount > 1 ) ? "s" : ""] for [( multi_payment ) ? "each shot of [gun.name]" : "usage license of [gun.name]"]?", "Weapon Purchase", list("Yes", "No"))
 		if(!user.can_perform_action(src))
 			active_prompt = FALSE
 			return FALSE
 		switch(license_request)
 			if("Yes")
-				if(credit_card_details.adjust_money(-payment_amount, "Firing Pin: Gun License"))
+				if(multi_payment)
+					gun_owners += credit_card_details
+					to_chat(user, span_notice("Gun rental terms agreed to, have a secure day!"))
+				else if(credit_card_details.adjust_money(-payment_amount, "Firing Pin: Gun License"))
 					if(pin_owner)
-						pin_owner.registered_account.adjust_money(payment_amount, "Firing Pin: Gun License Bought")
-					gun_owners += user
+						pin_owner.adjust_money(payment_amount, "Firing Pin: Gun License Bought")
+					gun_owners += credit_card_details
 					to_chat(user, span_notice("Gun license purchased, have a secure day!"))
-					active_prompt = FALSE
-					return FALSE //we return false here so you don't click initially to fire, get the prompt, accept the prompt, and THEN the gun
-				to_chat(user, span_warning("ERROR: User balance insufficent for successful transaction!"))
-				return FALSE
+				else 
+					to_chat(user, span_warning("ERROR: User balance insufficent for successful transaction!"))
+				active_prompt = FALSE
+				return FALSE //we return false here so you don't click initially to fire, get the prompt, accept the prompt, and THEN the gun
 			if("No")
```

Comment:
```suggestion
			if("No", null)
```

In order to respect any potential value from the tgui_alert above (e.g. someone closing out of the alert). may as well do it now to be nice